app/main.py

```diff
@@ -1,3 +1,74 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: float) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(
+                km=self.km + other
+            )
+
+        return Distance(
+            km=self.km + other.km
+        )
+
+    def __iadd__(self, other: float) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
+
+            return self
+        self.km += other.km
+
+        return self
+
+    def __mul__(self, other: float) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(
+                km=self.km * other
+            )
+
+    def __truediv__(self, other: float) -> Distance:
+        if not isinstance(other, Distance):
+            print(self.km / other)
+            return Distance(
+                km=round(self.km / other, 2)
+            )
+
+    def __lt__(self, other: float) -> bool:
+        if not isinstance(other, Distance):
+            return self.km < other
+
+        return self.km < other.km
+
+    def __gt__(self, other: float) -> bool:
```

Comment:
The `__gt__` method should handle comparisons with both `Distance` instances and numeric types. Ensure `other` is a numeric type when it's not a `Distance`.